File: test-varela.js
const { createClient } = require('@supabase/supabase-js');
require('dotenv').config({ path: '.env.local' });

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY;
const supabase = createClient(supabaseUrl, supabaseKey);

async function run() {
    const { data: profiles, error: pErr } = await supabase.from('profiles').select('*');
    if (pErr) console.log(pErr);

    console.log("Profile Columns:", profiles.length > 0 ? Object.keys(profiles[0]) : "Empty Array");

    const varela = profiles.find(p => p.last_name && p.last_name.toLowerCase().includes('varela'));

    if (varela) {
        const { data: stats } = await supabase.from('performance_stats').select('*').eq('player_id', varela.id);

        console.log(`\n--- VARELA STATS LOG ---`);
        console.log(`Stats Rows Found: ${stats.length}`);

        // Per 90 Calculations
        const per90s = stats.filter(s => s.minutes > 0).map(s => {
            const p90 = s.total_distance * (90 / s.minutes);
            console.log(`Raw: ${s.total_distance}m in ${s.minutes}min -> Per90: ${p90.toFixed(1)}m`);
            return p90;
        });

        console.log(`\nMax Per 90 Target Value: ${Math.max(...per90s).toFixed(1)}m`);

        // Rolling Avg + 5% Calculation
        const distValues = stats.map(s => s.total_distance);
        const sum = distValues.reduce((a, b) => a + b, 0);
        const avg = sum / distValues.length;
        console.log(`Rolling Average + 5% Target Value: ${(avg * 1.05).toFixed(1)}m`);
    } else {
        console.log("no varela found in DB.");
    }
}
run();

File: test_varela.py
import os
import requests
from dotenv import load_dotenv

# Load env variables from Next.js local config
load_dotenv('.env.local')

url = os.getenv('NEXT_PUBLIC_SUPABASE_URL') + '/rest/v1/profiles?select=*'
headers = {
    'apikey': os.getenv('NEXT_PUBLIC_SUPABASE_ANON_KEY'),
    'Authorization': 'Bearer ' + os.getenv('NEXT_PUBLIC_SUPABASE_ANON_KEY')
}

profiles_response = requests.get(url, headers=headers).json()
if isinstance(profiles_response, list):
    varela = next((p for p in profiles_response if p.get('last_name') and 'varela' in p['last_name'].lower()), None)
else:
    print('Raw Error Response:', profiles_response)
    varela = None
print(varela)

if varela and 'id' in varela:
    print(f"Found Varela. ID: {varela['id']}")
    
    stat_url = os.getenv('NEXT_PUBLIC_SUPABASE_URL') + f"/rest/v1/performance_stats?select=total_distance,minutes&player_id=eq.{varela['id']}"
    stats = requests.get(stat_url, headers=headers).json()
    
    print(f"Varela Match Rows Found: {len(stats)}")
    for s in stats:
        print(f"   - Distance: {s['total_distance']}m, Minutes: {s['minutes']}")
        
    per_90s = [s['total_distance'] * (90/s['minutes']) for s in stats if s['minutes'] > 0]
    print(f"\nCalculated Per-90 Array: {[round(p, 1) for p in per_90s]}")
    if per_90s:
        print(f"Max Per-90 Historical Target: {round(max(per_90s), 1)}")
        
    total_dist = sum(s['total_distance'] for s in stats)
    avg_dist = total_dist / len(stats) if stats else 0
    print(f"Rolling Average + 5% Target: {round(avg_dist * 1.05, 1)}")
else:
    print("Could not find Varela in profiles array")
